spec/lib/miq_ldap_spec.rb

```diff
@@ -119,4 +119,20 @@
     data = "\001\005\000\000\000\000\000\005\025\000\000\000+\206\301\364y\307\r\302=\336p\216\237\004\000\000"
     expect(MiqLdap.sid_to_s(data)).to eq("S-1-5-21-4106323499-3255682937-2389761597-1183")
   end
+
+  it 'returns a hostname when a hostname is availble and does not disable verify mode' do
+    allow(TCPSocket).to receive(:gethostbyname).and_return(["testhostname", "aliases", "type", "192.168.252.20"])
+    allow(TCPSocket).to receive(:new)
+    ldap = MiqLdap.new(:host => ["testhostname", "localhost", "dummy", @host])
+    expect(ldap.ldap.host).to eq("testhostname")
+    expect(ldap.ldap.instance_variable_get(:@encryption).try(:has_key_path?, :tls_options, :verify_mode)).to be_falsey
```

Comment:
would prefer check against OpenSSL::SSL::VERIFY_NONE vs just existence of key.

Reply:
But in this case, the key path shouldn't even exist, right?

Reply:
@abellotti This line confirms the :verify_mode does ***not*** exists.

Reply:
So I think we should keep the existing and add:

```ruby
expect(ldap.ldap.instance_variable_get(:@encryption).fetch_path(:tls_options, :verify_mode)).to_not eq(OpenSSL::SSL::VERIFY_NONE)
```

Reply:
Why add any more checks against that key?  If it doesn't exist, it can't equal anything.

Reply:
@bdunne Good question. I just try to make the reviewers happy :)

Reply:
those reviewers :), I'd just remove the test on line 128 as the one on 129 is semantically clearer (for me), Thanks.

Reply:
I disagree, I think 128 is clearer :)

Reply:
Can I go with both to make both @abellotti and @bdunne happy? :)

Reply:
At the moment it is true that the only value used for verify_mode by this code is `OpenSSL::SSL::VERIFY_NONE` but what if in the future it's updated to support a different `verify_mode`?
With this in mind I think line 129 is the better way to code the spec.

@bdunne Does this perspective change your mind or do you still feel line 128 is clearer?

Reply:
Is this information accessible via a method call to `ldap.ldap` rather than `instance_variable_get?  If so, that would be the best case.  Then we can verify that the method has the correct value.

Reply:
@bdunne there is an #encryption method but it is not an accessor. It has a different function. That's sort of what got us to this approach.